adcc/adc_pp/state2state_transition_dm.py

```diff
@@ -20,12 +20,119 @@
 ## along with adcc. If not, see <http://www.gnu.org/licenses/>.
 ##
 ## ---------------------------------------------------------------------
+from adcc import block as b
 from adcc.AdcMethod import AdcMethod
+from adcc.functions import einsum
 from adcc.AmplitudeVector import AmplitudeVector
+from adcc.OneParticleOperator import OneParticleOperator
 
 import libadcc
 
-DISPATCH = {}  # None implemented
+
+def s2s_tdm_adc0(mp, amplitude_l, amplitude_r, intermediates):
+    x1 = amplitude_l["s"]
+    y1 = amplitude_r["s"]
+
+    dm = OneParticleOperator(mp, is_symmetric=False)
+    dm[b.oo] = -einsum('ja,ia->ij', x1, y1)
+    dm[b.vv] = einsum('ia,ib->ab', x1, y1)
+    return dm
+
+
+def s2s_tdm_adc2(mp, amplitude_l, amplitude_r, intermediates):
+    dm = s2s_tdm_adc0(mp, amplitude_l, amplitude_r, intermediates)
+
+    x1 = amplitude_l["s"]
+    y1 = amplitude_r["s"]
+    x2 = amplitude_l["d"]
+    y2 = amplitude_r["d"]
+
+    t2 = mp.t2(b.oovv)
+    p0_ov = mp.mp2_diffdm[b.ov]
+    p0_oo = mp.mp2_diffdm[b.oo]
+    p0_vv = mp.mp2_diffdm[b.vv]
+    p1_oo = dm[b.oo].evaluate()  # ADC(1) tdm
+    p1_vv = dm[b.vv].evaluate()  # ADC(1) tdm
+
+    rx1 = einsum('ijab,jb->ia', t2, x1)
+    ry1 = einsum('ijab,jb->ia', t2, y1)
+
+    dm[b.oo] = (
+        p1_oo - 2.0 * einsum('ikab,jkab->ij', y2, x2) + (
+            + 0.5 * (
+                einsum('ik,kj->ij', p1_oo, p0_oo)
+                + einsum('ik,kj->ij', p0_oo, p1_oo)
+            )
```

Comment:
This (and in other places) is an explicit symmetrise. Any reason you do it this way?

Reply:
Just tried to remove the explicit symmetrise, but got failing tests even when using the `symmetrise` function on the result 😭

Reply:
No what I meant is replace the explicitly coded `symmetrise` by an appropriate call to `symmetrise` (to save lines of code).

Reply:
Yes, I know what you meant. 👍  Still, I could not figure out the correct spot to add the `symmetrise` call... There is already an overall `symmetrise` around the expression, but this seemingly does not suffice, and doing e.g. `einsum('ik,kj->ij', p1_oo, p0_oo).symmetrise()` also led to failing tests.

Reply:
You kept in mind that unlike adcman we have an 0.5 inside our definition of symmetrise? (Sorry for asking these stupid questions, but it's just stuff I often forget).

Reply:
Yeah, so to make it clear, what I did was to convert this expression
```Python
0.5 * (einsum('ik,kj->ij', p1_oo, p0_oo) + einsum('ik,kj->ij', p0_oo, p1_oo))
```
to
```Python
einsum('ik,kj->ij', p1_oo, p0_oo).symmetrise()
```

Reply:
That's strange, but ok ... let's not bother with this until we have a working version.